File: 01_Arrays_and_Hashing/03_two_sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> m = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int val = target - nums[i];
            if (m.containsKey(val)) {
                return new int[] { i, m.get(val) };
            } else {
                m.put(nums[i], i);
            }
        }

        return new int[0];
    }
}

File: 01_Arrays_and_Hashing/03_two_sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        
        m = {}

        for index, val in enumerate(nums):
            if (target - val) in m.keys():
                return [index, m[target-val]]
            else:
                m[val] = index

        return []